aggregator/src/test/java/org/jenkinsci/plugins/workflow/steps/input/InputStepFileParameterTest.java

```diff
@@ -0,0 +1,149 @@
+import org.apache.commons.httpclient.methods.PostMethod;
+import org.apache.commons.httpclient.methods.multipart.FilePart;
+import org.apache.commons.httpclient.methods.multipart.MultipartRequestEntity;
+import org.apache.commons.httpclient.methods.multipart.Part;
+import org.apache.commons.httpclient.methods.multipart.StringPart;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowExecution;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.jenkinsci.plugins.workflow.support.steps.input.InputAction;
+import org.jenkinsci.plugins.workflow.support.steps.input.InputStepExecution;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * @author <a href="mailto:tom.fennelly@gmail.com">tom.fennelly@gmail.com</a>
+ */
+public class InputStepFileParameterTest {
+
+    @Rule
+    public JenkinsRule jenkinsRule = new JenkinsRule();
+    
+    @Test
+    @Issue("JENKINS-29289")
+    public void test_input_file_param() throws Exception {
+        
+        // Job setup:
+        // It's a script that looks for an file as input. It:
+        //   1. makes sure the file doesn't exist in workspace before the input step is reached
+        //   2. manually POSTs a file to the input action to trigger the proceed flow (yeah, could maybe use the webclient etc etc ... too finicky)
+        //   3. makes sure the file does exist in workspace after the input step has execute 
+        
+        WorkflowJob foo = jenkinsRule.jenkins.createProject(WorkflowJob.class, "foo");
+        foo.setDefinition(new CpsFlowDefinition("node {\n" +
+                "    stage 's1'\n" +
+                "    sh 'echo Hi'\n" +
+                "\n" +
+                "    def file = new File(pwd(), 'text.txt');\n" +
+                "\n" +
+                "    file.delete();\n" +
+                "    if (file.exists()) {\n" +
+                "        throw new IllegalStateException(\"File should not exist\");\n" +
+                "    } else {\n" +
+                "        echo \"${file} is NOT in workspace. OK!!\"\n" +
+                "    }\n" +
+                "\n" +
+                "    input message: 'Load input file', parameters: [[$class: 'FileParameterDefinition', description: '', name: 'text.txt']]\n" +
+                "\n" +
+                "    // File should now be in the workspace... \n" +
+                "\n" +
+                "    if (!file.exists()) {\n" +
+                "        throw new IllegalStateException(\"File should exist\");\n" +
+                "    } else {\n" +
+                "        echo \"${file} is in workspace. OK!!\"\n" +
+                "    }\n" +
+                "\n" +
+                "}"));
```

Comment:
Hmmmm.... do you really think so? I prefer being able to see it all in one place if possible.